### PR TITLE
Use Random instead of UUID to generate randomized prefix for echange spooling

### PR DESCRIPTION
**Pull request by linzebing**
## Description

> Is this change a fix, improvement, new feature, refactoring, or other?

Improvement

> Is this a change to the core query engine, a connector, client library, or the SPI interfaces? (be specific)

trino-exchange-filesystem

> How would you describe this change to a non-technical end user or system administrator?

Improves the performance of randomized prefix generation.

## Documentation

(x) No documentation is needed.
( ) Sufficient documentation is included in this PR.
( ) Documentation PR is available with #prnumber.
( ) Documentation issue #issuenumber is filed, and can be handled later.

## Release notes

(x) No release notes entries required.
( ) Release notes entries required with the following suggested text:

```markdown
# Section
* Fix some things. ({issue}`issuenumber`)
```

### CODE REVIEW

**Comment by martint** [comment]
What's the motivation? Please add a note to the commit message.

File: plugin/trino-exchange-filesystem/src/main/java/io/trino/plugin/exchange/filesystem/FileSystemExchange.java [hunk 1]
```diff
@@ -65,6 +65,8 @@
         implements Exchange
 {
     private static final Pattern PARTITION_FILE_NAME_PATTERN = Pattern.compile("(\\d+)_(\\d+)\\.data");
+    private static final char[] ALPHANUMERIC = "abcdefghijklmnopqrstuvwzyz0123456789".toCharArray();
```
Comment (arhimondr) [inline plugin/trino-exchange-filesystem/src/main/java/io/trino/plugin/exchange/filesystem/FileSystemExchange.java:68]: `ALPHANUMERIC` usually implies `[A-Za-z0-9]`. How about calling it `RANDOMIZED_PREFIX_ALPHABET`?

**Review by arhimondr — APPROVED** [review]
LGTM % comment

File: plugin/trino-exchange-filesystem/src/main/java/io/trino/plugin/exchange/filesystem/FileSystemExchange.java [hunk 2]
```diff
@@ -43,10 +43,10 @@
 import java.util.Map;
 import java.util.Optional;
 import java.util.Set;
-import java.util.UUID;
 import java.util.concurrent.CompletableFuture;
 import java.util.concurrent.ConcurrentHashMap;
 import java.util.concurrent.ExecutorService;
+import java.util.concurrent.ThreadLocalRandom;
```
Comment (findepi) [inline plugin/trino-exchange-filesystem/src/main/java/io/trino/plugin/exchange/filesystem/FileSystemExchange.java:49]: from cmt msg

> such that the first two chars has as many combinations as possible.

why?

Comment (linzebing) [reply plugin/trino-exchange-filesystem/src/main/java/io/trino/plugin/exchange/filesystem/FileSystemExchange.java:49]: Ah already merged. This is mainly for S3. Users can talk to AWS to set up manual partitioning for S3 buckets, and they do that base on the first n chars. In our case, it's the first 2.

**Pull request merged**